pkgs/applications/science/electronics/kicad/default.nix

```diff
@@ -28,7 +32,15 @@
 , withI18n ? true
 }:
 
-assert ngspiceSupport -> libngspice != null;
+assert withNgspice -> libngspice != null;
+assert stdenv.lib.assertMsg (!ngspiceSupport)
+  "nspiceSupport is deprecated. Use withNgspice instead.";
+assert stdenv.lib.assertMsg (!oceSupport)
+  "oceSupport is deprecated. Use withOCE instead.";
+assert stdenv.lib.assertMsg (!scriptingSupport)
+  "scriptingSupport is deprecated. Use withScripting instead.";
+assert stdenv.lib.assertMsg (!withOCCT)
+  "withOCCT is deprecated. Use withOCC instead.";
```

Comment:
On a 2nd thought, now it hits me that we didn't have a deprecation period :laughing: . Personally I'm in favor of doing small incompatible changes like this without warning, but at least we should comply to the definition of the words we are using. So maybe (example):

```
Please use `withOCCT` instead of `withOCC`
```

Or:

```
`withOCCT` was renamed to `withOCC` for the sake of consistency with upstream cmake options
```

And:

```
`scriptingSupport` was renamed to `withScripting` for the sake of consistency with other kicad nix arguments.
```

Reply:
You mean take out these deprecation assertions? I'm happy to do that if you think it's better. If we leave them in, how long before I should come back and take them all out?

And don't worry about the many rounds of edits, better to get this right the first time than to have to come back and fix it later :)

Reply:
I meant only to change the message.

Reply:
Please leave the assertions as are.

Reply:
Got it. Ignore that last force push. I still need to add the release notes.